File: civitasruby/lib/sorpresa_por_jugador.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
module Civitas
class SorpresaPorJugador < Sorpresa
  def initialize(texto,valor)
    super(texto)
    @valor = valor
  end
  
  def aplicar_a_jugador(actual,todos)
     if(jugador_correcto(actual,todos))
        super(actual,todos)
        a=SorpresaPagarCobrar.new("Pagan todos los jugadores menos el no afectado ",@valor * -1)
        for i in 0...todos.size
          if(i!=actual)
            a.aplicar_a_jugador(i,todos)
          end
        end
        b=SorpresaPagarCobrar.new("Recibe dinero de todos los jugadores",@valor*(todos.size-1))
        b.aplicar_a_jugador(actual,todos)
    end
  end
  
  def to_string
    a = "Por Jugador"
    return a
  end
  public_class_method :new
end
end

File: civitasjava/src/Civitas/SorpresaPorJugador.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author info
 */
public class SorpresaPorJugador extends Sorpresa{
    private int valor;
    
    SorpresaPorJugador(int valor, String texto){
        super(texto);
        this.valor = valor;
    }
    
    @Override
    public void aplicarAJugador(int actual, ArrayList<Jugador> todos){
        if(jugadorCorrecto(actual, todos)){
            informe(actual, todos);
            SorpresaPagarCobrar a = new SorpresaPagarCobrar(valor * -1, 
            "Pagan todos menos el que no es afectado");
            
            for(int i=0; i<todos.size(); i++){
                if(i!=actual)
                    a.aplicarAJugador(i, todos);
            }
            
            SorpresaPagarCobrar b = new SorpresaPagarCobrar(
                    valor*(todos.size()-1), "Recibe dinero de todos");
            }
    }
    
    @Override
    public String toString(){
        return "SorpresaAPlicarAJugador{ valor: " +valor +super.toString() +"}";
    }
    
    
    
}
